tests/unit/project_loader/grammar/test_compound_statement.py

```diff
@@ -0,0 +1,243 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2017 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import testtools
+from testtools.matchers import Equals
+from unittest.mock import patch
+
+import snapcraft
+from snapcraft.internal.project_loader import grammar
+import snapcraft.internal.project_loader.grammar._on as on
+import snapcraft.internal.project_loader.grammar._to as to
+import snapcraft.internal.project_loader.grammar._compound as compound
+
+from . import GrammarBaseTestCase
+
+
+class CompoundStatementGrammarTestCase(GrammarBaseTestCase):
+
+    scenarios = [
+        ('on amd64', {
+            'on': 'on amd64',
+            'to': 'to armhf',
+            'body': ['foo'],
+            'else_bodies': [],
+            'host_arch': 'x86_64',
+            'expected_packages': {'foo'}
+        }),
+        ('on i386', {
+            'on': 'on amd64',
+            'to': 'to armhf',
+            'body': ['foo'],
+            'else_bodies': [],
+            'host_arch': 'i686',
+            'expected_packages': set()
+        }),
+        # The fact that this doesn't pass is probably a bug
```

Comment:
what does this mean?

Reply:
Ah, it was a note to self and is now removed.